src/bloom.cpp

```diff
@@ -254,8 +254,8 @@ void CRollingBloomFilter::insert(const std::vector<unsigned char>& vKey)
         if (nGeneration == 4) {
             nGeneration = 1;
         }
-        uint64_t nGenerationMask1 = -(uint64_t)(nGeneration & 1);
-        uint64_t nGenerationMask2 = -(uint64_t)(nGeneration >> 1);
+        uint64_t nGenerationMask1 = (uint64_t)(-(int64_t)(nGeneration & 1));
+        uint64_t nGenerationMask2 = (uint64_t)(-(int64_t)(nGeneration >> 1));
```

Comment:
This works around the error, but I don't think it's very elegant. What about just formulating it explicitly:
```
        uint64_t nGenerationMask1 = ~(uint64_t)(nGeneration & 1) + 1;
        uint64_t nGenerationMask2 = ~(uint64_t)(nGeneration >> 1) + 1;
```